hs_composite_resource/models.py

```diff
@@ -254,6 +227,20 @@ def _recreate_xml_docs_for_folder(self, folder, check_metadata_dirty=False):
             for aggregation in aggregations:
                 aggregation.create_aggregation_xml_documents()
 
+    def get_aggregation_by_dataset_pathname(self, aggregation_name):
+        """Get an aggregation that matches the aggregation dataset_name specified by *dataset_name*
+        :param  name: name (aggregation path) of the aggregation to find
+        :return an aggregation object if found
+        :raises ObjectDoesNotExist if no matching aggregation is found
+        """
+        for aggregation in self.logical_files:
+            # remove the last slash in aggregation_name if any
+            if aggregation.aggregation_name == aggregation_name:
+                return aggregation
```

Comment:
no lash slash is checked to be removed if any as stated in the comment

Reply:
it's a leftover comment, removed